packages/sdk/src/permission.ts

```diff
@@ -55,14 +58,18 @@ export interface ChainPermissions {
     canGrant: boolean
 }
 
-export const isPublicPermissionQuery = (query: InternalPermissionQuery): query is PublicPermissionQuery => {
-    return (query as PublicPermissionQuery).public === true
+export const isPublicPermissionQuery = (query: InternalPermissionQuery): query is InternalPublicPermissionQuery => {
+    return (query as InternalPublicPermissionQuery).public === true
```

Comment:
I'm guessing `InternalUserPermissionQuery` doesn't have the `public` key that's why the cast (`… as …`). If that's true then the following is sufficient for such a gate:

```ts
return `public` in query
```

No casting!

Reply:
The current approach is better safer as it errors if `public` isn't a key in the expected interface (InternalPublicPermissionQuery). The `... in query` approach wouldn't error. 

Maybe relevant if we'll rename this field in some future refactoring.

Reply:
I see. Cheers!